apiserver/facades/agent/uniter/lxdprofile.go

```diff
@@ -21,8 +21,6 @@ import (
 // charms with lxd profiles is completely different from the approach in
 // NewLXDProfileAPI.
 
-//go:generate go run go.uber.org/mock/mockgen -package mocks -destination mocks/lxdprofile.go github.com/juju/juju/apiserver/facades/agent/uniter LXDProfileBackend,LXDProfileMachine,LXDProfileUnit
```

Comment:
Are you sure it's better to store all mocks files in the same folder? In previous PRs I prefer to place generated mocks in separate subfolder `mocks`.

Reply:
I really think that having mocks in different packages promotes the wrong reuse. It's not your mocks it's mine.

Reply:
Yes, the `mocks` sub-package pattern was only ever needed to avoid circular imports, which hints at poor layout, potentially poor interface locations.

They should just go into the same directory by default and be named _x_mock_test.go_.